src/Areas/Marketplace/Services/MarketplaceService.cs

```diff
@@ -0,0 +1,168 @@
+
+namespace AzureMcp.Areas.Marketplace.Services;
+
+public class MarketplaceService(ITenantService tenantService)
+    : BaseAzureService(tenantService), IMarketplaceService
+{
+    private const int TokenExpirationBuffer = 300;
+    private const string ManagementApiBaseUrl = "https://management.azure.com";
+    private const string ApiVersion = "2023-01-01-preview";
+    private static readonly HttpClient s_sharedHttpClient = new HttpClient();
+
+    private string? _cachedAccessToken;
+    private DateTimeOffset _tokenExpiryTime;
+
+    /// <summary>
+    /// Retrieves a single private product (offer) for a given subscription.
+    /// </summary>
+    /// <param name="productId">The ID of the product to retrieve.</param>
+    /// <param name="subscription">The Azure subscription ID.</param>
+    /// <param name="includeStopSoldPlans">Include stop-sold or hidden plans.</param>
+    /// <param name="language">Product language (default: en).</param>
+    /// <param name="market">Product market (default: US).</param>
+    /// <param name="lookupOfferInTenantLevel">Check against tenant private audience.</param>
+    /// <param name="includeHiddenPlans">Include hidden plans.</param>
+    /// <param name="planId">Filter by plan ID.</param>
+    /// <param name="skuId">Filter by SKU ID.</param>
+    /// <param name="includeServiceInstructionTemplates">Include service instruction templates.</param>
+    /// <param name="partnerTenantId">Partner tenant ID.</param>
+    /// <param name="pricingAudience">Pricing audience.</param>
+    /// <param name="objectId">AAD user ID.</param>
+    /// <param name="altSecId">Alternate Security ID (for MSA users).</param>
+    /// <param name="tenant">Optional. The Azure tenant ID for authentication.</param>
+    /// <param name="retryPolicy">Optional. Policy parameters for retrying failed requests.</param>
+    /// <returns>A JSON node containing the product information.</returns>
+    /// <exception cref="ArgumentException">Thrown when required parameters are missing or invalid.</exception>
+    /// <exception cref="Exception">Thrown when parsing the product response fails.</exception>
+    public async Task<JsonNode?> GetProduct(
+        string productId,
+        string subscription,
+        bool? includeStopSoldPlans = null,
+        string? language = null,
+        string? market = null,
+        bool? lookupOfferInTenantLevel = null,
+        bool? includeHiddenPlans = null,
+        string? planId = null,
+        string? skuId = null,
+        bool? includeServiceInstructionTemplates = null,
+        string? partnerTenantId = null,
+        string? pricingAudience = null,
+        string? objectId = null,
+        string? altSecId = null,
+        string? tenant = null,
+        RetryPolicyOptions? retryPolicy = null)
+    {
+        ValidateRequiredParameters(productId, subscription);
+
+        string productUrl = BuildProductUrl(subscription, productId, includeStopSoldPlans, language, market,
```

Comment:
Do we not have an Azure SDK for marketplace?  If not, we should look at creating on.  Regardless, we should see if we can use the Azure Core to get HttpRetries etc.